test/wallet-http.js

```diff
@@ -0,0 +1,650 @@
+
+    const state = Resource.decode(Buffer.from(ns.data, 'hex'));
+
+    const resource = await wallet.getResource(ns.name);
+    const res = Resource.fromJSON(resource);
+
+    assert.deepEqual(state, res);
+  });
+
+  it('should create a renewal', async () => {
+    await wallet.createOpen({
+      name: name
+    });
+
+    const {treeInterval} = network.names;
+    for (let i = 0; i < treeInterval + 1; i++)
+      await nclient.execute('generatetoaddress', [1, cbAddress]);
+
+    await sleep(100);
+
+    await wallet.createBid({
+      name: name,
+      bid: 1000,
+      lockup: 2000
+    });
+
+    const {biddingPeriod} = network.names;
+    for (let i = 0; i < biddingPeriod + 1; i++)
+      await nclient.execute('generatetoaddress', [1, cbAddress]);
+
+    await sleep(100);
+
+    await wallet.createReveal({
+      name: name
+    });
+
+    const {revealPeriod} = network.names;
+    for (let i = 0; i < revealPeriod + 1; i++)
+      await nclient.execute('generatetoaddress', [1, cbAddress]);
+
+    await sleep(100);
+
+    await wallet.createUpdate({
+      name: name,
+      data: {
+        text: ['foobar']
+      }
+    });
+
+    // mine up to the earliest point in which a renewal
+    // can be submitted, a treeInterval into the future
+    for (let i = 0; i < treeInterval + 1; i++)
+      await nclient.execute('generatetoaddress', [1, cbAddress]);
+
+    await sleep(100);
+
+    const json = await wallet.createRenewal({
+      name
+    });
+
+    const updates = json.outputs.filter(({covenant}) => covenant.type === types.RENEW);
+    assert.equal(updates.length, 1);
+  });
+
+  it('should create a transfer', async () => {
+    await wallet.createOpen({
+      name: name
+    });
+
+    const {treeInterval} = network.names;
+    for (let i = 0; i < treeInterval + 1; i++)
+      await nclient.execute('generatetoaddress', [1, cbAddress]);
+
+    await sleep(100);
+
+    await wallet.createBid({
+      name: name,
+      bid: 1000,
+      lockup: 2000
+    });
+
+    const {biddingPeriod} = network.names;
+    for (let i = 0; i < biddingPeriod + 1; i++)
+      await nclient.execute('generatetoaddress', [1, cbAddress]);
+
+    await sleep(100);
+
+    await wallet.createReveal({
+      name: name
+    });
+
+    const {revealPeriod} = network.names;
+    for (let i = 0; i < revealPeriod + 1; i++)
+      await nclient.execute('generatetoaddress', [1, cbAddress]);
+
+    await sleep(100);
+
+    await wallet.createUpdate({
+      name: name,
+      data: {
+        text: ['foobar']
+      }
+    });
+
+    for (let i = 0; i < treeInterval + 1; i++)
+      await nclient.execute('generatetoaddress', [1, cbAddress]);
+
+    await sleep(100);
+
+    const {receiveAddress} = await wallet.getAccount(accountTwo);
+
+    const json = await wallet.createTransfer({
+      name,
+      address: receiveAddress
+    });
+
+    const xfer = json.outputs.filter(({covenant}) => covenant.type === types.TRANSFER);
+    assert.equal(xfer.length, 1);
```

Comment:
Do you want to check the namestate here and confirm the transfer worked? Might be good just for extra validity check of the transfer tx.